From 07c3b93b709a601398ce5e06d48f215d097ed959 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Tue, 24 Sep 2013 16:27:20 +0200
Subject: [PATCH] logind: return -EINVAL when PID is wrong

dbus-send --print-reply --system --dest=org.freedesktop.login1
/org/freedesktop/login1 org.freedesktop.login1.Manager.GetUserByPID
uint32:0
causes
systemd-logind[29843]: Assertion 'pid >= 1' failed at
src/login/logind.c:938, function manager_get_user_by_pid(). Aborting.
---
 src/login/logind.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/login/logind.c b/src/login/logind.c
index 4ef92b8..afc737c 100644
--- a/src/login/logind.c
+++ b/src/login/logind.c
@@ -914,9 +914,11 @@ int manager_get_session_by_pid(Manager *m, pid_t pid, Session **session) {
         int r;
 
         assert(m);
-        assert(pid >= 1);
         assert(session);
 
+        if (pid < 1)
+                return -EINVAL;
+
         r = cg_pid_get_unit(pid, &unit);
         if (r < 0)
                 return r;
@@ -935,9 +937,11 @@ int manager_get_user_by_pid(Manager *m, pid_t pid, User **user) {
         int r;
 
         assert(m);
-        assert(pid >= 1);
         assert(user);
 
+        if (pid < 1)
+                return -EINVAL;
+
         r = cg_pid_get_slice(pid, &unit);
         if (r < 0)
                 return r;
